pkg/agent/controller/networkpolicy/priority.go

```diff
@@ -26,125 +26,231 @@ import (
+	case (insertionPoint > PriorityTopCNP || nextVacant == nil) && lastVacant != nil:
+		return pa.siftPrioritiesDown(insertionPoint-uint16(1), *lastVacant, p)
+	case nextVacant != nil && lastVacant != nil:
+		costSiftUp := *nextVacant - insertionPoint
+		costSiftDown := insertionPoint - *lastVacant - uint16(1)
+		if costSiftUp < costSiftDown {
+			return pa.siftPrioritiesUp(insertionPoint, *nextVacant, p)
+		} else {
+			return pa.siftPrioritiesDown(insertionPoint-uint16(1), *lastVacant, p)
 		}
-		pa.priorityMap[priority] = computedPriority
+	default:
+		return nil, map[uint16]uint16{}, fmt.Errorf("no available Openflow priority left to insert priority %v", p)
+	}
+}
+
+// siftPrioritiesUp moves all consecutive occupied ofPriorities and corresponding Priorities up by one ofPriority,
+// starting from the insertionPoint. It also assigns the freed ofPriority to the input Priority.
+func (pa *priorityAssigner) siftPrioritiesUp(insertionPoint, nextVacant uint16, p types.Priority) (*uint16, map[uint16]uint16, error) {
+	priorityReassignments := map[uint16]uint16{}
+	if insertionPoint >= nextVacant {
+		return nil, priorityReassignments, fmt.Errorf("failed to determine the range for sifting priorities")
+	}
+	for i := nextVacant; i > insertionPoint; i-- {
+		p, _ := pa.ofPriorityMap[i-uint16(1)]
+		pa.updatePriorityAssignment(i, p)
+		priorityReassignments[i-uint16(1)] = i
+		klog.V(2).Infof("Original priority %v now needs to be re-assigned %v", i-uint16(1), i)
+	}
+	pa.updatePriorityAssignment(insertionPoint, p)
+	return &insertionPoint, priorityReassignments, nil
+}
+
+// siftPrioritiesDown moves all consecutive occupied ofPriorities and corresponding Priorities down by one ofPriority,
+// starting from the insertionPoint. It also assigns the freed ofPriority to the input Priority.
+func (pa *priorityAssigner) siftPrioritiesDown(insertionPoint, lastVacant uint16, p types.Priority) (*uint16, map[uint16]uint16, error) {
+	priorityReassignments := map[uint16]uint16{}
+	if insertionPoint <= lastVacant {
+		return nil, priorityReassignments, fmt.Errorf("failed to determine the range for sifting priorities")
 	}
-	return &newPriority, priorityUpdates, nil
+	for i := lastVacant; i < insertionPoint; i++ {
+		p, _ := pa.ofPriorityMap[i+uint16(1)]
+		pa.updatePriorityAssignment(i, p)
+		priorityReassignments[i+uint16(1)] = i
+		klog.V(2).Infof("Original priority %v now needs to be re-assigned %v", i+uint16(1), i)
+	}
+	pa.updatePriorityAssignment(insertionPoint, p)
+	return &insertionPoint, priorityReassignments, nil
 }
 
 // GetOFPriority retrieves the OFPriority for the input Priority to be installed,
 // and returns installed priorities that need to be re-assigned if necessary.
 func (pa *priorityAssigner) GetOFPriority(p types.Priority) (*uint16, map[uint16]uint16, error) {
-	ofPriority, exists := pa.priorityMap[p]
-	if !exists {
-		return pa.syncPriorityZone(p)
+	noPriorityReassignments := map[uint16]uint16{}
```

Comment:
Is `noPriorityReassignments` used to represent empty map?
I think you can just return nil for no reassignment case as you only read the result.
> var m map[string]int
> Map types are reference types, like pointers or slices, and so the value of m above is nil; it doesn't point to an initialized map. A nil map behaves like an empty map when reading, but attempts to write to a nil map will cause a runtime panic; don't do that.

Reply:
Done.